src/public/index.ts

```diff
@@ -54,8 +59,12 @@ function main() {
   const readButton = document.getElementById("readButton") as HTMLElement;
   const showExampleButton = document.getElementById("showExampleButton") as HTMLElement;
   fileInput.addEventListener("change", uploader.showFile.bind(uploader));
-  readButton.addEventListener("click", () => {
-    uploader.readFile();
+  readButton.addEventListener("click", async () => {
+    await uploader.readFile();
```

Comment:
would it make sense to make readFile return an object with all those properties below (sourceKey, propertiesValue, etc)? I see it as more 'natural' than creating getters to those properties (not blocking just a suggestion)

Reply:
good point, I was taking to do it. not sure why I changed my mind or  I think just forgot to do the change.